Source/CombatExtended/CombatExtended/Comps/CompSuppressable.cs

```diff
@@ -78,6 +78,30 @@ private float SuppressionThreshold
             }
         }
 
+        private float breakThresholdMajorCached;
+        private int breakThresholdMajorTickCheck;
+        private float BreakThresholdMajorCached(Pawn pawn)
+        {
+            if (breakThresholdMajorTickCheck == 0 || Find.TickManager.TicksGame > (breakThresholdMajorTickCheck + 30))
```

Comment:
Can this not be a simple modulo check against `Find.TickManager.TicksGame`?
```suggestion
            if ((Find.TickManager.TicksGame + 17) % 30 == 0)
```
17 as a random offset to not block the round % 30 too much.

IIRC the game even has a helper method for this exact purpose, although I'm not near the code to tell you which it is.